code/drasil-theory/Theory/Drasil/Theory.hs

```diff
@@ -16,8 +16,9 @@ class Theory t where
   quantities    :: Lens' t [QuantityDict]
   operations    :: Lens' t [ConceptChunk] -- FIXME: Should not be Concept
   defined_quant :: Lens' t [QDefinition]
-  invariants    :: Lens' t [Relation]
+  invariants    :: Lens' t [DisplayExpr]  -- TODO: temporary hack until designed, previously `Lens' t [Relation]`
   defined_fun   :: Lens' t [QDefinition]
+  display_exprs :: t -> [DisplayExpr]     -- TODO: temporary hack until designed
```

Comment:
This area needs a little bit of design work.

Should the `invariants` be a list of Relations? If we only intend to display them, then this change might be "okay".

The `display_exprs` is meant to be a list of all the related defined quantities, functions, and invariants.

Would it be okay to merge the defined_fun (functions) and defined_quants (quantities)? It doesn't appear that we use the defined_fun list yet, and we can later filter based on the Space if need be (once we have function spaces).

Reply:
The various pieces of a theory were defined from semantic considerations. We did not yet use all its features.

Once we do have types, yes, we can merge `defined_fun` and `defined_quants`.